plugins/kubernetes-backend/src/routes/resourcesRoutes.ts

```
@@ -0,0 +1,113 @@
+    }
+
+    function getBearerToken(header?: string): string | undefined {
+      return header?.match(/Bearer\s+(\S+)/i)?.[1];
+    }
+    const token = getBearerToken(req.headers.authorization);
+
+    if (!token) {
+      throw new AuthenticationError('No Backstage token');
+    }
+
+    const entity = await catalogClient.getEntityByRef(entityRef, {
+      token: token,
+    });
+
+    if (!entity) {
+      throw new InputError(
+        `Entity ref missing, ${stringifyEntityRef(entityRef)}`,
+      );
+    }
+    return entity;
+  };
+
+  router.post('/resources/workloads/query', async (req, res) => {
+    const entity = await getEntityByReq(req);
+
+    try {
+      const response = await objectsProvider.getKubernetesObjectsByEntity({
+        entity,
+        auth: req.body.auth,
+      });
+      res.json(response);
+    } catch (e) {
+      logger.error(String(e), {
+        action: '/resources/workloads',
+        entityRef: stringifyEntityRef(entity),
+      });
+      res.status(500).json({ error: e.message });
+    }
+  });
+
+  router.post('/resources/custom/query', async (req, res) => {
+    const entity = await getEntityByReq(req);
+
+    if (!req.body.customResources) {
+      throw new InputError('customResources is a required field');
+    } else if (!Array.isArray(req.body.customResources)) {
+      throw new InputError('customResources must be an array');
+    } else if (req.body.customResources.length === 0) {
+      throw new InputError('at least 1 customResource is required');
+    }
+
+    try {
+      const response = await objectsProvider.getCustomResourcesByEntity({
+        entity,
+        customResources: req.body.customResources,
+        auth: req.body.auth,
+      });
+      res.json(response);
+    } catch (e) {
+      logger.error(String(e), {
+        action: '/resources/custom',
+        entityRef: stringifyEntityRef(entity),
+      });
+      res.status(500).json({ error: e.message });
```

same here, just let the error bubble up